pkg/debuginfo/manager.go

```diff
@@ -270,6 +303,43 @@ func (di *Manager) debuginfoSrcPath(ctx context.Context, buildID string, objFile
 	return objFile.Path
 }
 
+func (di *Manager) upload(ctx context.Context, uploadInstructions *debuginfopb.UploadInstructions, r io.Reader) error {
+	switch uploadInstructions.UploadStrategy {
+	case debuginfopb.UploadInstructions_UPLOAD_STRATEGY_GRPC:
+		return uploadViaGRPC(ctx, di.debuginfoClient, uploadInstructions, r)
+	case debuginfopb.UploadInstructions_UPLOAD_STRATEGY_SIGNED_URL:
+		return uploadViaSignedURL(ctx, uploadInstructions.SignedUrl, r)
+	case debuginfopb.UploadInstructions_UPLOAD_STRATEGY_UNSPECIFIED:
+		return fmt.Errorf("upload strategy unspecified, must set one of UPLOAD_STRATEGY_GRPC or UPLOAD_STRATEGY_SIGNED_URL")
+	default:
+		return fmt.Errorf("unknown upload strategy: %v", uploadInstructions.UploadStrategy)
+	}
+}
+
+func uploadViaGRPC(ctx context.Context, debuginfoClient debuginfopb.DebuginfoServiceClient, uploadInstructions *debuginfopb.UploadInstructions, r io.Reader) error {
+	_, err := parcadebuginfo.NewGrpcUploadClient(debuginfoClient).Upload(ctx, uploadInstructions, r)
```

Comment:
How costly is it to create this client? Shall we store it as part of the manager as `Uploader`?

Reply:
this has basically no cost, it's just a wrapper around the grpc client https://github.com/parca-dev/parca/blob/5541ffa12d1e0f66783092972db93f9a350f207b/pkg/debuginfo/client.go#L49-L51